Objective-C/CBLCollection.mm

```diff
@@ -951,4 +952,30 @@ - (FLSliceResult) emptyFLSliceResult: (CBLDatabase*)db {
     return result;
 }
 
+- (nullable CBLQueryIndex*) getIndex:(nonnull NSString*)name error: (NSError**)error {
+    CBLAssertNotNil(name);
+    
+    CBL_LOCK(_mutex) {
+        if (![self checkIsValid: error])
+            return nil;
+        
+        C4Error* c4err = {};
+        CBLStringBytes iName(name);
+        
+        C4Index* c4index = c4coll_getIndex(_c4col, iName, c4err);
+        if (c4err) {
```

Comment:
Should be `!c4Index && err.code !=0 `.

Reply:
or 

if (!c4Index) {
   if (error.code != 0) {
      convertError(*c4err, error);
   }
   return nil;
}

return [[CBLQueryIndex alloc] initWithIndex: c4index name: name collection: self];